### PR TITLE
Copied SAI profile handler from SONiC project

### PR DESCRIPTION
**Pull request by MaxPolovyi**
Copied SONiC implementation of `handle_profile_map`, `ndi_profile_get_next_value`, `ndi_profile_get_value`.
The default SAI profile file name is `/etc/opx/sai-profile`.
The format of sai-profile file is the same as used in SONiC and PTF test framework.
Since this api is intended to be used from systemd service, environment variable is the only viable way to change profile filename in runtime.

### CODE REVIEW

File: inc/opx/nas_ndi_utils.h [hunk 1]
```diff
@@ -88,6 +88,13 @@ static inline t_std_error ndi_utl_mk_std_err (enum e_std_error_subsystems sub,
              STD_ERR_MK (sub, e_std_err_code_FAIL, 0);
 }
 
+t_std_error handle_profile_map(sai_switch_profile_id_t profile_id,
+                               const char* profile_file_name);
```
Comment (pkarashchenko) [inline inc/opx/nas_ndi_utils.h]: Please change to "const char *profile_file_name"

Comment (pkarashchenko) [inline inc/opx/nas_ndi_utils.h]: Please change to "const char *"

File: src/nas_ndi_utils.cpp [hunk 2]
```diff
@@ -227,3 +233,93 @@ bool ndi_to_sai_vlan_stats(ndi_stat_id_t ndi_id, sai_vlan_stat_t * sai_id){
...
+    auto it = profile_map[profile_index].find(variable);
+    if (it == profile_map[profile_index].end()) {
+        NDI_INIT_LOG_INFO("%s: NULL", variable);
+        return NULL;
+    }
+    NDI_INIT_LOG_TRACE("%s: %s", variable, it->second.c_str());
+    return it->second.c_str();
+}
+
+static std::vector<std::map<std::string, std::string>::iterator> profile_iter;
```
Comment (pkarashchenko) [inline src/nas_ndi_utils.cpp]: @MaxPolovyi Please move this to top of the file

File: src/nas_ndi_utils.cpp [hunk 3]
```diff
@@ -227,3 +233,93 @@ bool ndi_to_sai_vlan_stats(ndi_stat_id_t ndi_id, sai_vlan_stat_t * sai_id){
     *sai_id = it->second;
     return true;
 }
+
+static std::map<sai_switch_profile_id_t, size_t> profile_id_map;
+static std::vector<std::map<std::string, std::string>> profile_map;
+
+const char* ndi_profile_get_value(sai_switch_profile_id_t profile_id, const char* variable) {
```
Comment (pkarashchenko) [inline src/nas_ndi_utils.cpp]: @MaxPolovyi Please change to "const char *"

File: src/nas_ndi_utils.cpp [hunk 4]
```diff
@@ -227,3 +233,93 @@ bool ndi_to_sai_vlan_stats(ndi_stat_id_t ndi_id, sai_vlan_stat_t * sai_id){
...
+        NDI_INIT_LOG_INFO("%s: NULL", variable);
+        return NULL;
+    }
+    NDI_INIT_LOG_TRACE("%s: %s", variable, it->second.c_str());
+    return it->second.c_str();
+}
+
+static std::vector<std::map<std::string, std::string>::iterator> profile_iter;
+
+int ndi_profile_get_next_value(sai_switch_profile_id_t profile_id, const char** variable, const char** value)
```
Comment (pkarashchenko) [inline src/nas_ndi_utils.cpp]: Please change to "const char **"

File: src/nas_ndi_utils.cpp [hunk 5]
```diff
@@ -227,3 +233,93 @@ bool ndi_to_sai_vlan_stats(ndi_stat_id_t ndi_id, sai_vlan_stat_t * sai_id){
...
+    }
+    *variable = profile_iter[profile_index]->first.c_str();
+    *value = profile_iter[profile_index]->second.c_str();
+    NDI_INIT_LOG_TRACE("key: %s:%s", *variable, *value);
+    profile_iter[profile_index]++;
+    return 0;
+}
+
+t_std_error handle_profile_map(sai_switch_profile_id_t profile_id, 
+                               const char* profile_file_name) {
```
Comment (pkarashchenko) [inline src/nas_ndi_utils.cpp]: Please change to "const char *"

File: src/nas_ndi_utils.cpp [hunk 6]
```diff
@@ -227,3 +233,93 @@ bool ndi_to_sai_vlan_stats(ndi_stat_id_t ndi_id, sai_vlan_stat_t * sai_id){
...
+            NDI_INIT_LOG_INFO("not found '=' in line %s", line.c_str());
+            continue;
+        }
+        std::string key = line.substr(0, pos);
+        std::string value = line.substr(pos + 1);
+        profile_map[profile_id][key] = value;
+
+        NDI_INIT_LOG_INFO("insert: %s:%s", key.c_str(), value.c_str());
+    }
+}
```
Comment (pkarashchenko) [inline src/nas_ndi_utils.cpp]: @MaxPolovyi Please file new line at the end of file

File: src/nas_ndi_utils.cpp [hunk 7]
```diff
@@ -227,3 +233,93 @@ bool ndi_to_sai_vlan_stats(ndi_stat_id_t ndi_id, sai_vlan_stat_t * sai_id){
...
+    if (profile_iter.size() <= profile_index) {
+        size_t profile_iter_size = profile_iter.size();
+        profile_iter.resize(profile_index + 1);
+        for (int i = profile_iter_size; i <= profile_iter.size(); ++i) {
+            profile_iter[i] = profile_map[i].begin();
+        }
+    }
+    std::ifstream profile(profile_map_file);
+    if (!profile.is_open()) {
+        NDI_INIT_LOG_ERROR("failed to open profile map file: %s : %s", profile_map_file.c_str(), strerror(errno));
```
Comment (pkarashchenko) [inline src/nas_ndi_utils.cpp]: @MaxPolovyi I think that if not file is found the warning message should be logged.

**Comment by MaxPolovyi** [comment]
@pkarashchenko , I've resolved your comments.

**Pull request merged**